src/opensearch_sdk_py/transport/stream_input.py

```diff
@@ -161,8 +161,8 @@ def read_array_size(self) -> int:
         if array_size > 2**31:
             raise Exception(f"array length must be <= to {2**31} but was: {array_size}")
 
-        if array_size < 0:
-            raise Exception(f"array size must be positive but was: {array_size}")
+        # if array_size < 0:
+        #    raise Exception(f"array size must be positive but was: {array_size}")
```

Comment:
Was probably copied from Java, we can delete this.

Reply:
Yeah, I'm sure it was there to protect overflow, had a mental note to also look at the 2^31 limit and the commented line below it as well, at some point.